p5js/NeuralNetwork/NeuralNetwork_basics/sketch.js

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2018 ml5
+//
+// This software is released under the MIT License.
+// https://opensource.org/licenses/MIT
+
+/* ===
+ml5 Example
+Image classification using MobileNet and p5.js
+This example uses a callback pattern to create the classifier
+=== */
+let nn;
+
+function setup() {
+  // Options for Neural Network
+  const options = {
+    inputs: 3,
+    outputs: 2,
+    task:'regression',
+    // activationOutput: 'sigmoid',
+    // activationHidden: 'sigmoid'
+  };
+  // Create Neural Network
+  nn = ml5.neuralNetwork(options);
+
+  trainModel();
+
+  // Train the model
+  let trainBtn = createButton('Train Model');
+  trainBtn.position(10, 50);
+  trainBtn.mousePressed(function () {
+    trainModel();
+  });
+
+  // Predict
+  let predictBtn = createButton('Predict');
+  predictBtn.position(10, 70);
+  predictBtn.mousePressed(function () {
+    predict();
+  });
+
+  // Save and download the model
+  let saveBtn = createButton('Save Model');
+  saveBtn.position(10, 90);
+  saveBtn.mousePressed(function () {
+    neuralNetwork.save();
+  });
+
+  // Load the model from local files
+  let loadLocalBtn = createButton('Load the model from local files');
+  loadLocalBtn.position(10, 110);
+  loadLocalBtn.mousePressed(function () {
+    neuralNetwork.load('model/model.json', function () {
+      console.log('Model Loaded!');
+    });
+  });
+
+  // Load model
+  let loadBtn = select('#load');
+  loadBtn.changed(function () {
+    neuralNetwork.load(loadBtn.elt.files, function () {
+      console.log('Model Loaded!');
+    });
+  });
+}
+
+function trainModel() {
+  // Add training data
+  // const training_input = [-0.6, 1, 0.25];
+  // const training_target = [0.3, 0.9];
+  let a,b,c;
+  let training_target;
+  for (let i = 0; i < 500; i++) {
+    if(i % 2){
+       a = Math.random(0,0.16);
+       b = Math.random(0.16,0.32);
+       c = Math.random(0.32,0.5);
+       training_target = [0,0]
+    }else {
+      a = Math.random(0.5,0.66);
+      b = Math.random(0.66,0.82);
+      c = Math.random(0.82,1);
+      training_target = [1,1]
+    }
+    
+    const training_input = [a, b, c];
+    // const training_target = [0, 1];
+
+    nn.data.addData({
```

Comment:
(Take this with a grain of salt b/c I haven't done a full review of everything just poking around and tinkering!)

Ideally when the data is generated in the code itself without a CSV that has headers or a JSON file with named properties we would allow the user to just pass in vanilla arrays. So, for example, this would become:

```javascript
// arrays are in variables
nn.data.addData(training_input, training_target);
// or to be more clear:
nn.data.addData([0.2, 0.7. -0.1], [1, 0]);
```

I started working on implementing this but thought maybe it's better discussed first before I pull apart a lot of the`NeuralNetworkData`code!

Reply:
Ok, in looking into this a bit more, I found something that perhaps could be adjusted to make the data class easier to work with ultimately?

The way the data is shaped originally is inverted from the way it ultimately ends up being shaped for `train()`. For example, with the titanic example, the data first appears in `NeuralNetworkData` as `[3, 916]`:

<img width="724" alt="Screen Shot 2019-09-20 at 6 53 20 PM" src="https://user-images.githubusercontent.com/191758/65363205-9f16c800-dbd8-11e9-927b-228ac1f1919e.png">
(output of: https://github.com/ml5js/ml5-library/blob/generic-neural-network/src/NeuralNetwork/NeuralNetworkData.js#L72)

But the actual input tensor for training is (as it should be) `[916, 3]`:

<img width="545" alt="Screen Shot 2019-09-20 at 6 54 35 PM" src="https://user-images.githubusercontent.com/191758/65363252-dbe2bf00-dbd8-11e9-8238-358f51e6915d.png">

This confused me as I was trying out some ideas for adjustments to ml5-library based on the comment above.

Reply:
Re arrays:
+ Yes! totally. I also think adding in blank arrays is the best way to go. I used the named json objects to best match how our current examples are currently structured with the intention to make this more flexible. 

Re data structure:
+ Maybe I misunderstood how the tf.tensor.max() and .min() functions work, but I thought that maybe the data needed to be restructured so that those functions would be correctly applied. If this is not correct, then we can totally save those steps! Maybe needs a second look.